### ISSUE TITLE
Separation of reply notifications from comment notifications, and direct notification linking to specific comment/reply

### ISSUE THREAD

**Issue by Carl2213** [humhub/humhub#4390]
### Is your feature request related to a problem? Please describe.
- When a reply notification is received, the original post is quoted. E.g. "Person commented on 'Hi'". No indication is given that this is a reply to a specific comment, and not a comment on the main post.
- When the notification is clicked, it takes the user to the main post. The user has to then discover where the comment/reply from the notification is. This is very confusing on posts with many comments. Sometimes, the comments are even hidden/collapsed after clicking the notification, adding more confusion.
- Comment/reply notifications are also grouped together. If Person A replied to my comment and then Person B commented on the original post, the notification will say "Person A and Person B commented on 'Hi'". This makes it very easy to miss replies to earlier comments because the notification seems like it's only for new comments at the end of the post.
### Describe the solution you'd like
- Notifications for replies should be separate from comment notifications. E.g. "Person replied to your comment".
- Notifications for comments/replies should link directly to the comment/reply, expanding comments/replies if necessary. Maybe the comment/reply in question can be highlighted briefly onload.
- If the notification is for a reply, quote the comment it is replying to, instead of quoting the original post. E.g. "Person replied to your comment 'Hello!'"

**Comment by Carl2213** [comment]
Separating comment and reply notification logic should also (maybe) fix #4316

**Comment by samsheridan** [comment]
This is desperately needed. I don't want to use nested comments on a production site until this is resolved.

**Comment by luke-** [comment]
Will enhancements #4316 and #4399 solve the problem?

**Comment by Carl2213** [comment]
Yes Luke, but #4316 (before the name change) referred to something else that I'm not sure if it would be fixed by separating reply notifications.

Users shouldn't get notified for replies to comments other than their own.

**Comment by buddh4** [comment]
Regarding the problem that a user first has to find the comment, we probably need at first a better detail with with a proper comment section. Currently even in the detail view (single entry stream) only a few comments are shown.

**Comment by tosintech-web** [comment]
How about notification of reply of comments? I think user should get notified of reply to their own comments but the original post.
